### PR TITLE
Update to go1.19

### PR DESCRIPTION
**Pull request by cpanato**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

- Update to go1.19

/assign @saschagrunert @palnabarun @xmudrii @Verolop 
cc @kubernetes/release-managers 

/hold fro any discussions and other updates (test-infra jobs)

#### Does this PR introduce a user-facing change?

```release-note
- Update to go1.19
```

### CODE REVIEW

File: Dockerfile-kubepkg-rpm [hunk 1]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM golang:1.18.5-buster AS builder
+FROM golang:1.19-buster AS builder
```
Comment (cpanato) [inline Dockerfile-kubepkg-rpm:15]: should we update to use `bullseye`?

File: Dockerfile-kubepkg [hunk 2]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM golang:1.18.5-buster AS builder
+FROM golang:1.19-buster AS builder
```
Comment (cpanato) [inline Dockerfile-kubepkg:15]: should we update to use `bullseye`?

File: Dockerfile-kubepkg-rpm (hunk 1 above)
Comment (saschagrunert) [reply Dockerfile-kubepkg-rpm:15]: Yeah let's do that in one step.

Comment (cpanato) [reply Dockerfile-kubepkg-rpm:15]: Do you mean in a follow up? or in this PR as well?

Comment (saschagrunert) [reply Dockerfile-kubepkg-rpm:15]: In this PR here

**Review by saschagrunert — APPROVED** [review]

**Comment by cpanato** [comment]
job update in test-infra: https://github.com/kubernetes/test-infra/pull/27291

**Review by palnabarun — APPROVED** [review]
/lgtm
/approve

/priority important-soon

**Pull request merged**